From 2051f94de3a24a17e4edcca4223b036c0db23500 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Sat, 11 Jan 2020 19:58:17 +0000
Subject: [PATCH] show.cpp: don't make CvWindows private

* with newer boost 1.69.0 it fails with:

In file included from /jenkins/mjansa/build-ros/webos-dashing-warrior/tmp-glibc/work/raspberrypi3-webos-linux-gnueabi/swri-opencv-util/3.0.3-1-r0/git/src/show.cpp:35:
/jenkins/mjansa/build-ros/webos-dashing-warrior/tmp-glibc/work/raspberrypi3-webos-linux-gnueabi/swri-opencv-util/3.0.3-1-r0/recipe-sysroot/usr/include/boost/serialization/singleton.hpp: In instantiation of 'boost::serialization::detail::singleton_wrapper<T>::singleton_wrapper() [with T = swri_opencv_util::CvWindows]':
/jenkins/mjansa/build-ros/webos-dashing-warrior/tmp-glibc/work/raspberrypi3-webos-linux-gnueabi/swri-opencv-util/3.0.3-1-r0/recipe-sysroot/usr/include/boost/serialization/singleton.hpp:171:47:   required from 'static T& boost::serialization::singleton<T>::get_instance() [with T = swri_opencv_util::CvWindows]'
/jenkins/mjansa/build-ros/webos-dashing-warrior/tmp-glibc/work/raspberrypi3-webos-linux-gnueabi/swri-opencv-util/3.0.3-1-r0/recipe-sysroot/usr/include/boost/serialization/singleton.hpp:193:28:   required from 'static T& boost::serialization::singleton<T>::get_mutable_instance() [with T = swri_opencv_util::CvWindows]'
/jenkins/mjansa/build-ros/webos-dashing-warrior/tmp-glibc/work/raspberrypi3-webos-linux-gnueabi/swri-opencv-util/3.0.3-1-r0/git/src/show.cpp:89:25:   required from here
/jenkins/mjansa/build-ros/webos-dashing-warrior/tmp-glibc/work/raspberrypi3-webos-linux-gnueabi/swri-opencv-util/3.0.3-1-r0/recipe-sysroot/usr/include/boost/serialization/singleton.hpp:147:24: error: 'swri_opencv_util::CvWindows::CvWindows()' is private within this context
     singleton_wrapper(){
                        ^
/jenkins/mjansa/build-ros/webos-dashing-warrior/tmp-glibc/work/raspberrypi3-webos-linux-gnueabi/swri-opencv-util/3.0.3-1-r0/git/src/show.cpp:71:5: note: declared private here
     CvWindows() {}
     ^~~~~~~~~

Upstream-Status: Pending

Signed-off-by: Martin Jansa <martin.jansa@lge.com>
---
 src/show.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/show.cpp b/src/show.cpp
index e85fb4867..e11e263ff 100644
--- a/src/show.cpp
+++ b/src/show.cpp
@@ -67,8 +67,8 @@ namespace swri_opencv_util
 #else
     friend class boost::serialization::detail::singleton_wrapper<CvWindows>;
 #endif
-  private:
     CvWindows() {}
+  private:
     boost::mutex mutex_;
     std::map<std::string, std::string> windows_;
   };
